src/App.tsx

```diff
@@ -1,42 +1,81 @@
-  { label: 'Recently Updated', value: 'updated', direction: 'desc' },
-  { label: 'Least Pull Requests', value: 'pulls', direction: 'asc' },
-  { label: 'Most Pull Requests', value: 'pulls', direction: 'desc' },
-  { label: 'Least Stars', value: 'stars', direction: 'asc' },
-  { label: 'Most Stars', value: 'stars', direction: 'desc' },
+  { label: "Least Recently Updated", value: "updated", direction: "asc" },
+  { label: "Recently Updated", value: "updated", direction: "desc" },
+  { label: "Least Pull Requests", value: "pulls", direction: "asc" },
+  { label: "Most Pull Requests", value: "pulls", direction: "desc" },
+  { label: "Least Stars", value: "stars", direction: "asc" },
+  { label: "Most Stars", value: "stars", direction: "desc" },
 ];
 
 function App() {
-  const { repos, loading, error, sortRepos, searchRepos } = useGitHub();
-  const [searchTerm, setSearchTerm] = useState('');
+  const { repos, loading, error, fetchRepos } = useGitHub();
+  const [searchTerm, setSearchTerm] = useState("");
   const [selectedSort, setSelectedSort] = useState<SortOption>(sortOptions[0]);
+  const filterState = useSelector((state: RootState) => state.filtering);
   const reposRedux = useSelector((state: RootState) => state.repo.value);
-  
-  // Take the Repos Returned from the Github Hook, and set them in Redux
   const dispatch = useDispatch();
-  dispatch(setReposInRedux(repos));
+
+  useEffect(() => {
+    dispatch(setReposInRedux(repos));
+  }, [dispatch, repos]);
+
+  const sortRepos = useCallback((repos: Repository[], option: SortOption) => {
+    return [...repos].sort((a, b) => {
+      const multiplier = option.direction === 'desc' ? -1 : 1;
+      
+      switch (option.value) {
+        case 'pulls':
+          return multiplier * ((a.pulls_count || 0) - (b.pulls_count || 0));
+        case 'updated':
+          return multiplier * (new Date(a.updated_at).getTime() - new Date(b.updated_at).getTime());
+        case 'stars':
+          return multiplier * (a.stargazers_count - b.stargazers_count);
+        default:
+          return 0;
+      }
+    });
+  },[]);
+
+  const searchRepos = useCallback((reposRedux: Repository[], searchTerm: string) => {
+    const term = searchTerm.toLowerCase();
+    return reposRedux.filter(repo => 
+      repo.name.toLowerCase().includes(term) ||
+      (repo.description?.toLowerCase().includes(term)) ||
+      (repo.readme?.toLowerCase().includes(term))
+    );
+  },[]);
+  const filterRepos = useCallback((repos: Repository[], filterState: FilterState) => {
+    return repos.filter(repo => {
+      const isArchived = !filterState.archiveCheckbox && repo.archived;
+      const isActive = !filterState.activeCheckbox && !repo.archived;
+      const isPublic = !filterState.publicCheckbox && repo.private;
+      const isPrivate = !filterState.privateCheckbox && !repo.private;
+      const isOrgSelected = !filterState.selectedOrgs[repo.organization || ''];
```

Comment:
what are the implications of an array with an empty string? Will that mean potentially something weird in the UI?

Reply:
I played around in `node` repl: It is possible that the ES would yield undefined from selectedOrgs. 

`!undefined === true` so that would mean that for a given organization, if it did NOT have any entry in the map, it would be selected. 

I'm okay with this behavior